File: Python/645_set_mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n = len(nums)
        seen = set()
        duplicate = 0

        for i in nums:
            if i in seen:
                duplicate = i
            else:
                seen.add(i)

        total = (n * (n + 1)) // 2
        missing = total - sum(seen)

        return [duplicate, missing]

File: Swift/645_set_mismatch.swift
class Solution {
    func findErrorNums(_ nums: [Int]) -> [Int] {
        let n = nums.count
        var seen = Set<Int>()
        var duplicate = 0

        for i in nums {
            if seen.contains(i) {
                duplicate = i
            } else {
                seen.insert(i)
            }
        }

        let sumSeen = seen.reduce(0, +)
        let total = (n * (n + 1 )) / 2
        let missing = total - sumSeen

        return [duplicate, missing]
    }
}